res/ari/resource_endpoints.c

```diff
@@ -307,3 +308,130 @@ void ast_ari_endpoints_send_message_to_endpoint(struct ast_variable *headers,
 	send_message(msg_to, args->from, args->body, variables, response);
 	ast_variables_destroy(variables);
 }
+
+static void send_refer(const char *to, const char *from, const char *refer_to, int to_self, struct ast_variable *variables, struct ast_ari_response *response)
+{
+	struct ast_variable *current;
+	struct ast_refer *refer;
+	int res = 0;
+
+	if (ast_strlen_zero(to)) {
+		ast_ari_response_error(response, 400, "Bad Request",
+			"To must be specified");
+		return;
+	}
+
+	refer = ast_refer_alloc();
+	if (!refer) {
+		ast_ari_response_alloc_failed(response);
+		return;
+	}
+
+	res |= ast_refer_set_from(refer, "%s", from);
```

Comment:
If some of these are optional, do we even need to call the ast_refer if they are empty?

Reply:
I'm not sure if I understand correctly. Do you mean I should do a `strlen`/NULL check of  `from`,`to`,... and only then call `ast_refer_set_from` etc. if necessary?

Reply:
!ast_strlen_zero specifically, and yes

Reply:
Makes sense, done. I'm also ignoring the return code since these always return 0.